### PR TITLE
[inductor] support unbacked symint divisors in vars_and_sizes

### PR DESCRIPTION
**Pull request by ColinPeppler**
Scenario:
```
>>> nodes
IterationRangesEntry(
    x2, 
    divisor=192*u0 + 192576, 
    length=s1, 
    (xindex//(192*u0 + 192576)), 
    {x0: 192, x1: u0 + 1003, x2: s1, x3: 192*s1*u0 + 192576*s1, x4: 192*u0 + 192576})
IterationRangesEntry(
    x1, 
    divisor=192, 
    length=u0 + 1003, 
    ModularIndexing(xindex, 192, u0 + 1003), 
    {x0: 192, x1: u0 + 1003, x2: s1, x3: 192*s1*u0 + 192576*s1, x4: 192*u0 + 192576})
IterationRangesEntry(
    x0, 
    divisor=1, 
    length=192, 
    ModularIndexing(xindex, 1, 192), 
    {x0: 192, x1: u0 + 1003, x2: s1, x3: 192*s1*u0 + 192576*s1, x4: 192*u0 + 192576})
```

Think about whether using fallback is safe here. I think it's safe because the divisor of one IterationRangesEntry should be the product of the lengths of the preceding IterationRangesEntry? Unless, one of the lengths divides by an unbacked symint?

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130595

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by aakhundov** [comment]
I guess, it should be fine? Assuming that the unbacked SymInt is a fixed non-negative value at runtime, replacing it with a (positive) fallback value should not affect the sorting order? Let's also ask @ezyang to be sure.

**Review by aakhundov — APPROVED** [review]

File: test/inductor/test_unbacked_symints.py [hunk 1]
```diff
@@ -252,6 +252,23 @@ def fn(x, w, repeats, is_bmm):
         expected = fn(*example_inputs)
         torch.testing.assert_close(actual, expected)
 
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
+    def test_unbacked_range_tree_divisor(self, device):
+        def fn(x, num):
+            u0 = num.item()
+            torch._check_is_size(u0)
+            zeros = torch.zeros(u0, device=device, dtype=torch.int)
```
Comment (ColinPeppler) [inline test/inductor/test_unbacked_symints.py:260]: not the best test, since it only mocks one unbacked symint divisor

**Review by ezyang — APPROVED** [review]

**Comment by aakhundov** [comment]
@ColinPeppler I tried restarting the failed builds (assuming the failures are transient), but they failed again. Maybe rebase?

**Pull request closed without merging**